base/gmp.jl

```diff
@@ -346,23 +346,27 @@ for (fJ, fC) in ((:-, :neg), (:~, :com))
     end
 end
 
-function <<(x::BigInt, c::Int)
-    c < 0 && throw(DomainError())
+function <<(x::BigInt, c::UInt)
```

Comment:
Possibly; I don't know whether `ccall` checks. 64-bit Linux definitely has this problem. That problem was there before, though.

I don't know a good solution -- shifting a `BigInt` left by a large number should work "in principle", but will in practice run out of memory. So an error message is probably the best we can do anyway.